public/viewjs/metrics.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Metrics Javascript
+ */
+
+/* Charting */
+var dataPoints = [];
+$("#metrics-table tbody tr").each(function () {
+	var self = $(this);
+	var label = self.find("td:eq(0)").attr('data-chart-label');
+	var value = Number(self.find("td:eq(1)").attr('data-chart-value'));
+	var dataPoint = { label: label, y: parseFloat((Math.round(value * 100) / 100).toFixed(2))};
+	dataPoints.push(dataPoint);
+});
+
+var options = {
+	exportEnabled: true,
+	legend:{
+		horizontalAlign: "center",
+		verticalAlign: "bottom"
+	},
+	data: [{
+		type: "pie",
+		showInLegend: true,
+		toolTipContent: "<b>{label}</b>: ${y} (#percent%)",
+		indexLabel: "{label}",
+		legendText: "{label} (#percent%)",
+		indexLabelPlacement: "outside",
+		valueFormatSTringt: "#,##0.##",
+		dataPoints: dataPoints
+	}]
+};
+
+// needed for recursionCount error
+recursionCount=0;
+$("#metrics-chart").CanvasJSChart(options);
+
+/* DataTables */
+var metricsTable = $('#metrics-table').DataTable({
+	"columnDefs": [
+		{ "type": "num", "targets": 1 }
+	]
+});
+$('#metrics-table tbody').removeClass("d-none");
+metricsTable.columns.adjust().draw();
+
+/* DateRangePicker */
+const urlParams = new URLSearchParams(window.location.search);
+
+var start_date = moment().startOf("month").format('MM/DD/YYYY');
+var end_date = moment().endOf("month").format('MM/DD/YYYY');
+
+if (urlParams.get('start_date')) start_date = moment(urlParams.get('start_date')) ;
+if (urlParams.get('end_date')) end_date = moment(urlParams.get('end_date'));
+
+$('#daterange-filter').daterangepicker({
+	showDropdowns: true,
+	startDate: start_date,
+	endDate: end_date,
+	locale: {
+		"format": 'MM/DD/YYYY',
+		"firstDay": 1
```

Comment:
Should respect [`CALENDAR_FIRST_DAY_OF_WEEK`](https://github.com/grocy/grocy/blob/91700e7dae4b35c32c2483ce0837a714df87a398/config-dist.php#L25-L28) (and maybe [`CALENDAR_SHOW_WEEK_OF_YEAR`](https://github.com/grocy/grocy/blob/91700e7dae4b35c32c2483ce0837a714df87a398/config-dist.php#L30-31)) as any other place a calendar is shown.

Reply:
Do you have any quick examples how to pull from the declared config the local data but in javascript?

Reply:
`Grocy.CalendarFirstDayOfWeek` for example [here](https://github.com/grocy/grocy/blob/91700e7dae4b35c32c2483ce0837a714df87a398/views/layout/default.blade.php#L99) - any place where a calendar / date-time-picker is used uses that...

Reply:
Excellent, `daterangepicker` has an options flag for both. :heavy_check_mark: 

https://www.daterangepicker.com/#options